AISKU/src/AISku.ts

```diff
@@ -259,8 +261,40 @@ export class AppInsightsSku implements IApplicationInsights {
                     _self.emptyQueue();
                     _self.pollInternalLogs();
                     _self.addHousekeepingBeforeUnload(_self);
+
+                    _addUnloadHook(onConfigChange(cfgHandler, () => {
+                        if (!_throttleMgr){
+                            _throttleMgr = new ThrottleMgr(_core);
+                        }
+                        //  && !_config.throttleMgrCfg.disabled
+                        if (_throttleMgr && !_throttleMgr.isReady() && _config.extensionConfig && _config.extensionConfig[_cfgSyncPlugin.identifier]) {
+                            // set ready state to true will automaatically trigger flush()
+                            _throttleMgr.onReadyState(true);
+                        }
+
+                        if (!_iKeySentMessage){ //should not send again
+                            if (!_config.connectionString && !isFeatureEnabled("disableIkeyDeprecationMessage", _config)) {
+                                _throttleMgr.sendMessage( _eInternalMessageId.InstrumentationKeyDeprecation, "Instrumentation key support will end soon, see aka.ms/IkeyMigrate");
+                                _iKeySentMessage = true;
+                            }
+                        }
+                       
+                        if (!_cdnSentMessage){
+                            if (sdkSrc && sdkSrc.indexOf("az416426") != -1 && !isFeatureEnabled("disableCdnDeprecationMessage", _config)) {
+                                _throttleMgr.sendMessage( _eInternalMessageId.CdnDeprecation, "Support for this domain will end soon, https://aka.ms/JsActiveCdn");
```

Comment:
Using terms like "will end soon" is problematic in the SDK as "when" is "soon" if someone starts using this version of the SDK in 5 years time (in theory) "soon" was several years ago. It's better to be a little bit more generic and just redirect them to the `aka.ms` link which can include dates as the resulting page can include specific timeframes and additional wording.

As an example (for this and the others) it could be phrased with something like
"Cdn Usage Notice, see https://aka.ms/xxxxx"